cpp/src/arrow/record_batch.cc

```diff
@@ -393,7 +393,7 @@ Result<std::shared_ptr<RecordBatchReader>> RecordBatchReader::Make(
 RecordBatchReader::~RecordBatchReader() {
   auto st = this->Close();
   if (!st.ok()) {
-    ARROW_LOG(WARNING) << "Implicitly called RecordBatchReader::Close failed: " << st;
```

Comment:
Should it be 
“Implicitly called RecordBatchReader::Close failed”?

Reply:
I meant the warn message.

Reply:
I mostly wanted to maintain the message that was already there. But otherwise the message is meant to convey that Close was called implicitly (user did not explicitly close the reader) and that the status was non-successful.

I could change it but I don't have strong opinions on what would be better